src/pharmpy/model/statements.py

```diff
@@ -996,19 +998,26 @@ def central_compartment(self):
         Compartment(CENTRAL, amount=A_CENTRAL, doses=Bolus(AMT, admid=1))
         """
         try:
-            # E.g. TMDD models have more than one input
+            # E.g. TMDD models have more than one output
+            # TODO : Relying heavily on compartment naming for drug metabolite
             central = list(self._g.predecessors(output))[-1]
+            if central.name == "METABOLITE":
+                central = self.find_compartment("CENTRAL")
+                if central is None:
+                    raise ValueError('Cannot find central compartment')
         except IndexError:
             raise ValueError('Cannot find central compartment')
         return central
 
-    @property
-    def peripheral_compartments(self):
+    def peripheral_compartments(self, name=None):
```

Comment:
Since this is no longer a property it should change to a name with a verb. We have a similar situation for transits I think using `find_transit_compartments`. I like using name as the argument here.

Reply:
Changed the naming to find_peripheral_compartments() to be in line with the transits compartments